### PR TITLE
Classes

### PR DESCRIPTION
**Pull request by emiliazm**
Hello!

### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Ruby requirements
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### Project requirements
- Create class `Person` with  the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
  - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
- Create class `Student` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
- Create class `Teacher` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
  - Override `can_use_services?` so it always returns `true`.
- Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def can_use_services?
+    of_age || parent_permission
```
Comment (TedLivist) [inline person.rb]: - [x] I think you meant `of_age?` (see question mark) and `@parent_permission`, like so:

  ```
  of_age? || @parent_permission
  ```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initilize(age, classroom, name = 'Unknown', parent_permission: true)
```
Comment (TedLivist) [inline student.rb]: - [x] I think you misspelled `initialize` wrong here.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age
+
```
Comment (TedLivist) [inline person.rb]: - [x] **[OPTIONAL]** I would recommend you move these getters and setters to the top of the class.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @emiliazm,

Good job so far!
✔️ Good documentation
✔️ Classes are in different files
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by emiliazm** [comment]
Hello @TedLivist !
Thank you for your comments!
I made the changes that were required.

**Review by dasileker — APPROVED** [review]
### **Hi @emiliazm,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**